kratos/utilities/coordinate_transformation_utilities.h

```diff
@@ -717,10 +707,12 @@ class CoordinateTransformationUtils {
 	{
 		double Norm = 0;
 		for(typename TVectorType::iterator iComponent = rThis.begin(); iComponent < rThis.end(); ++iComponent)
-		Norm += (*iComponent)*(*iComponent);
-		Norm = sqrt(Norm);
-		for(typename TVectorType::iterator iComponent = rThis.begin(); iComponent < rThis.end(); ++iComponent)
-		*iComponent /= Norm;
+			Norm += (*iComponent)*(*iComponent);
+		Norm = std::sqrt(Norm);
+		if (Norm > std::numeric_limits<double>::epsilon()){
```

Comment:
What happens if the `Norm` is 0? IMO you should at least throw a warning.

Reply:
So, if the norm is zero, it means that all the normal vector component is zero. You should not divide by zero.

Reply:
This happens to me only for moving slip surface

Reply:
@bodhinandach could you please clarify when do you encounter this situation (zero normal) and what would you expect to happen in that case? I would think that a normal with (close to) zero norm is quite useless to determine the correct orientation of the slip condition so, if anything, I would either ignore the rotation (and not rotate the affected nodes) or throw an error.

Reply:
@jcotela So, in my case, I map normal vector from some type of "boundary" particles to the background node, for some slip boundary condition imposition, by using shape functions. These boundary particles can move through different background meshes. 

If, suppose that at one point the particle is exactly or very closely located to one node, and therefore, the shape functions to the other nodes go to zero, zero normals can be found in those nodes. But as it is a temporary variable, and the information will be deleted in the next time step, it is normal for me to encounter this issue, and therefore such warning is not needed.

Reply:
> So, if the norm is zero, it means that all the normal vector component is zero. You should not divide by zero.

I know that we shouldn't divide by zero. But if you assume that this situation might happen and you add the if statement, you should do something if the zero normal condition happens. 

I don't think having a zero normal to a surface is completely normal, taking into account that this could imply that your surface has zero area, so the user should at least be notified about this.

Reply:
> > So, if the norm is zero, it means that all the normal vector component is zero. You should not divide by zero.
> 
> I know that we shouldn't divide by zero. But if you assume that this situation might happen and you add the if statement, you should do something if the zero normal condition happens.
> 
> I don't think having a zero normal to a surface is completely normal, taking into account that this could imply that your surface has zero area, so the user should at least be notified about this.

Maybe in your case this is completely normal, but no for the other users of the `coordinate_transformation_utitlities`. Having said this, if we add the extra if statement, I'd definitively add at least a warning.

Reply:
Okay, then I will derive the `Normalize()` in my utilities, and add the warning in the core?

Reply:
> I don't think having a zero normal to a surface is completely normal, taking into account that this could imply that your surface has zero area, so the user should at least be notified about this.

In my case, it is not the real surface, it is the background volume which imposes the surface conditions.

Reply:
> Okay, then I will derive the `Normalize()` in my utilities, and add the warning in the core?

I'll derive it in your class and keep the core one as it is.

Reply:
so you want me to return the core, not to add the warning or throw an error?

Reply:
Yes, to avoid the `if` overhead. But let's wait for @jcotela 's opinion.

Reply:
I returned it back to what it is, if you want to add the error, I can also add it. Waiting for your decision.

Reply:
I agree that the accepting a zero normal as valid input should be specific of the derived implementation (since it would not work as expected in the general case). I think it would still be good to have some kind of check, even if it is just in debug, for the base class, but I would not block this PR for this.

Reply:
Maybe we can add the check only in debug mode. @bodhinandach if you want to do it, feel free to proceed. If not, we can do it in another PR.